browser/brave_content_browser_client.cc

```diff
@@ -295,12 +296,19 @@ void BraveContentBrowserClient::MaybeHideReferrer(
           GURL(),
           CONTENT_SETTINGS_TYPE_PLUGINS,
           brave_shields::kBraveShields);
+
+  // Top-level navigations get empty referrers (brave/brave-browser#3422).
+  GURL replacement_referrer_url;
+  if (!is_main_frame) {
+    // But iframe navigations get spoofed instead (brave/brave-browser#3988).
+    replacement_referrer_url = request_url.GetOrigin();
```

Comment:
@iefremov This is the same as #2070, which you've already reviewed, except for this extra bit. We no longer strip URLs for cross-origin iframe navigations (we continue to spoof instead). This means that the sum total of this change is to strip referrers only on top-level navigations.

Reply:
@fmarier Fine, I'm ok with the change. Just curious, does it weaken user privacy after all?